File: Chapter 02/MVCDesignPatternExample.java
class CorgiModel {
    private String name;

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}


class CorgiView {
    public void displayCorgiDetails(String corgiName) {
        System.out.println("Corgi Name: " + corgiName);
    }
}


class CorgiController {
    private CorgiModel model;
    private CorgiView view;

    public CorgiController(CorgiModel model, CorgiView view) {
        this.model = model;
        this.view = view;
    }

    public void updateCorgiName(String name) {
        model.setName(name);
    }

    public void displayCorgiDetails() {
        String corgiName = model.getName();
        view.displayCorgiDetails(corgiName);
    }
}


public class MVCDesignPatternExample {
    public static void main(String[] args) {
        CorgiModel model = new CorgiModel();
        CorgiView view = new CorgiView();
        CorgiController controller = new CorgiController(model, view);

        controller.updateCorgiName("Muzz");
        controller.displayCorgiDetails();
    }
} 

File: Chapter 02/MVCDesignPatternExample.py
class CorgiModel:
    def __init__(self):
        self.name = None
    def set_name(self, name):
        self.name = name
    def get_name(self):
        return self.name

class CorgiView:
    def display_corgi_details(self, corgi_name):
        print(f"Corgi Name: {corgi_name}")

class CorgiController:
    def __init__(self, model, view):
        self.model = model
        self.view = view
    def update_corgi_name(self, name):
        self.model.set_name(name)
    def display_corgi_details(self):
        corgi_name = self.model.get_name()
        self.view.display_corgi_details(corgi_name)

if __name__ == "__main__":
    model = CorgiModel()
    view = CorgiView()
    controller = CorgiController(model, view)
    controller.update_corgi_name("Muzz")
    controller.display_corgi_details()
